mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -300,6 +318,10 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group setParam */
+  @Since("2.2.0")
+  def setInitialModel(value: KMeansModel): this.type = set(initialModel, value)
```

Comment:
Can you elaborate this? I don't fully understand why we can not overwrite setting in set method. Thanks.